crates/biome_js_analyze/src/lint/nursery/use_qwik_valid_lexical_scope.rs

```diff
@@ -0,0 +1,100 @@
+use biome_analyze::{Ast, Rule, RuleDiagnostic, RuleDomain, RuleSource, declare_lint_rule};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_js_syntax::{AnyJsExpression, JsVariableStatement};
+use biome_rowan::AstNode;
+use biome_rule_options::use_qwik_valid_lexical_scope::UseQwikValidLexicalScopeOptions;
+
+declare_lint_rule! {
+    /// Disallow unserializable expressions in Qwik dollar ($) scopes.
+    ///
+    /// Ensures all captured values in Qwik components can be properly serialized for resumability.
+    /// See [Qwik Optimizer: Lexical Scope](https://qwik.dev/docs/advanced/optimizer/#lexical-scope) for proper usage patterns.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const print = (msg: string) => {
+    ///   console.log(msg);
+    /// };
```

Comment:
The documentation talks about unserialisable expressions inside the dollar scopes; however, this example doesn't use any `$`. Wouldn't it be more useful to show a bad example *that uses the `$`*?